src/syscalls/mod.rs

```diff
@@ -345,8 +347,152 @@ pub unsafe extern "C" fn sys_open(name: *const c_char, flags: i32, mode: u32) ->
 	}
 }
 
+#[hermit_macro::system]
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn sys_getcwd(buf: *mut c_char, size: usize) -> *const c_char {
+	let error = |e: Errno| {
+		e.set_errno();
+		null::<c_char>()
+	};
+
+	if size == 0 {
+		return error(Errno::Inval);
+	}
+
+	let cwd = fs::get_cwd();
+	if let Err(e) = cwd {
+		return error(e);
+	}
+
+	let Ok(cwd) = cwd else { unreachable!() };
+
+	let Ok(cwd) = CString::new(cwd) else {
+		return error(Errno::Noent);
+	};
+
+	if (cwd.count_bytes() + 1) > size {
+		return error(Errno::Range);
+	}
+
+	if buf.is_null() {
+		return cwd.into_raw();
+	}
```

Comment:
I don't think we should leak memory here, since this cannot be freed through `malloc`.

Let's do that in newlib if at all. Sorry for the back and forth. :D

Reply:
Let's not do it at all, since it's not mandatory and even discouraged in POSIX